[stable/airflow] Allow specifying nodePort for Web service

thirtyseven opened the pull request:
<!--
Thank you for contributing to helm/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/helm/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/helm/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/helm/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the
history. This will make it easier to identify new changes. The PR will be squashed
anyways when it is merged. Thanks.

For fast feedback, please @-mention maintainers that are listed in the Chart.yaml file.

Please make sure you test your changes before you push them. Once pushed, a CircleCI
will run across your changes and do some initial checks and linting. These checks run
very quickly. Please check the results. We would like these checks to pass before we
even continue reviewing your changes.
-->

#### What this PR does / why we need it:
Currently, it is not possible to request a specific NodePort in the Airflow chart.  Choosing a NodePort allows it to be included in the Webserver Base URL setting.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

Commit:
Allow specifying nodePort for Web service

Signed-off-by: Ted Kaplan <edward.kaplan@nielsen.com>

Commit:
Update README.md

Signed-off-by: Ted Kaplan <edward.kaplan@nielsen.com>

Aisuko commented on stable/airflow/templates/service-web.yaml:
```diff
@@ -22,4 +22,7 @@ spec:
     - name: web
       protocol: TCP
       port: {{ .Values.airflow.service.externalPort | default 8080 }}
+      {{- if .Values.airflow.service.nodePort }}
```
I have a consider on this, why we need to add a NodePort to the chart?   The NodePort should be random create by the controller. Also, the ingress can not transfer the `TCP` traffic, but I think `Istio` can help us do that.

gsemet replied:
Hi. I don't know the intend behind this, but freezing the nodeport allows to access to the pod more easily knowning the port, without using an ingress.

I think no matter the use case one should always use an ingress/loadbalancer/whatever, but for debug I can understand the use case.

I would prefer however a solution such as:
```yaml
{{- if (and (eq .Values.service.type "NodePort") (not (empty .Values.service.nodePort.http))) }}
      nodePort: {{ .Values.service.nodePort.http }}
      {{- end }}
```

Aisuko replied:
Yes, I believe that can be made more sense.

gsemet replied:
@thirtyseven can you update and update your PR pleae?

thirtyseven replied:
> Hi. I don't know the intend behind this, but freezing the nodeport allows to access to the pod more easily knowning the port, without using an ingress.
> 
> I think no matter the use case one should always use an ingress/loadbalancer/whatever, but for debug I can understand the use case.

Yes, that's the intent of this PR.  We are using this chart in a k8s environment that does not currently support ingress controllers, but does have Consul DNS for service discovery.  

I'll make the changes as requested.

Aisuko reviewed (commented):
Thanks for your PR and there may some a piece of advice, please have a look on this, thanks.

Commit:
Only use nodePort value if service type is NodePort

Signed-off-by: Ted Kaplan <edward.kaplan@nielsen.com>

Commit:
Bump version

Signed-off-by: Ted Kaplan <edward.kaplan@nielsen.com>

Commit:
Fix template variable names

Signed-off-by: Ted Kaplan <edward.kaplan@nielsen.com>

Commit:
Update nodePort template variable

Update airflow.service.nodePort to airflow.service.nodePort.http as
requested by @gsemet

Signed-off-by: Ted Kaplan <edward.kaplan@nielsen.com>
